core/includes/form.inc

```diff
@@ -2892,7 +2893,12 @@ function form_select_options($element, $choices = NULL) {
       else {
         $selected = '';
       }
-      $options .= '<option value="' . check_plain($key) . '"' . $selected . '>' . check_plain($choice) . '</option>';
+      if (isset($element[$key])) {
+        if (isset($element[$key]['#disabled']) && $element[$key]['#disabled'] == TRUE) {
```

Comment:
These two nested `if` statements can be combined and simplified into:

```php
      if (!empty($element[$key]['#disabled'])) {
        $disabled = ' disabled';
      }
```

In PHP, `!empty(TRUE)` always returns true, and doing `!empty` for an array element that is not set or doesn't exist always returns false.  

From the [PHP manual](https://www.php.net/manual/en/function.empty.php):
<blockquote>
A variable is considered empty if it does not exist or if its value equals false. empty() does not generate a warning if the variable does not exist.
</blockquote>